presets/ragengine/inference/inference.py

```diff
@@ -129,9 +132,35 @@ async def achat(
         """ Perform an asynchronous chat completion request. """
         try:
             base_model, base_max_len = self._get_default_model_info()
+
+            # 3 characters per token is a conservative approximation for English text and code. For other languages, this may vary.
+            content_token_approximation = sum(self.count_tokens(message.content) for message in messages if message.content) if messages else 0
+            logger.info(f"Content token approximation: {content_token_approximation} tokens for messages")
+
+            if content_token_approximation > LLM_CONTEXT_WINDOW:
+                logger.error(f"Content length exceeds context window size ({LLM_CONTEXT_WINDOW}). Please reduce the length of the messages.")
+                raise HTTPException(status_code=400, detail=f"Content length exceeds context window size ({LLM_CONTEXT_WINDOW}). Please reduce the length of the messages.")
+
+            # if max_tokens is not provided but content length is less than the context window, we can pass None for max_tokens to allow the model to decide
+            max_tokens = kwargs.get("max_tokens")
```

Comment:
and here we respect passing `None` for max tokens as well, but adjust it based off available tokens if need be